[ARM64] Switch to using Microsoft.Management.Infrastructure for WMI

snickler opened the pull request:
## Summary of the pull request
This PR resolves issues with the Arm64 compiled Dev Home that prevents WMI from querying properly, due to an issue with the System.Management package. 

![image](https://github.com/microsoft/devhome/assets/4016293/278ce133-a81d-4e6b-aa04-ec52911fbb50)


## References and relevant issues
#865 and #855

## Detailed description of the pull request / Additional comments
I switched the WMI querying calls to use the Cim classes in the Microsoft.Management.Infrastructure package to provide the same functionality as System.Management provides. 

## Validation steps performed
- Built locally and deployed to Arm64 device
- Opened Add Widgets pane and verified that I could retrieve data from CPU, Memory, GPU without any errors 


## PR checklist
- [X] Closes #855
- [ ] Tests added/passed
- [ ] Documentation updated

crutkas commented:
🔥  love it!

krschau commented on tools/SetupFlow/DevHome.SetupFlow/DevHome.SetupFlow.csproj:
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="LibGit2Sharp" Version="0.27.0-preview-0182" />
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
     <PackageReference Include="Microsoft.Internal.Windows.DevHome.Helpers" Version="1.0.*-*" />
+    <PackageReference Include="Microsoft.Management.Infrastructure" Version="2.0.0" />
```
Because you reference this package in SetupFlow.Common, I think you don't need to include it here.

krschau commented on tools/SetupFlow/DevHome.SetupFlow/Services/DevDriveManager.cs:
```diff
@@ -16,6 +16,7 @@
 using DevHome.SetupFlow.ViewModels;
 using DevHome.SetupFlow.Windows;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Management.Infrastructure;
```
Can remove System.Management here too.

snickler commented on tools/SetupFlow/DevHome.SetupFlow.Common/DevDriveFormatter/DevDriveFormatter.cs:
```diff
@@ -4,9 +4,14 @@
 using Windows.Win32;
 using Windows.Win32.Foundation;
 
+using static Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext;
```
Thanks @krschau  - I made all those changes requesteed

snickler commented:
@krschau  - It looks like it's having issues restoring the devhome dependencies and running into the Azure Artifacts Auth issue? 

https://github.com/microsoft/devhome/actions/runs/5181719073/jobs/9358542767#step:7:18

snickler commented:
@EricJohnson327 - any reason why the build process is trying `dotnet restore` first in front of `nuget restore` and duplicating the package restore process? It may be better to set the first task to continue on fail and then fail if the nuget restore can't restore the packages properly. 

https://github.com/microsoft/devhome/blob/7196b9a50d625e728b8681e7f626db467c87fbb0/.github/workflows/DevHome-CI.yml#L59-L67

EricJohnson327 commented:
> @EricJohnson327 - any reason why the build process is trying `dotnet restore` first in front of `nuget restore` and duplicating the package restore process? It may be better to set the first task to continue on fail and then fail if the nuget restore can't restore the packages properly.
> 
> https://github.com/microsoft/devhome/blob/7196b9a50d625e728b8681e7f626db467c87fbb0/.github/workflows/DevHome-CI.yml#L59-L67

@snickler I was noticing some flakiness with GitHub Actions builds.  I wanted to unblock PR validation for forked repos as fast as possible, so I just put both in for now.  I agree that this can be made better :)

snickler commented:
> @snickler It's weird that it seems to be still looking for the System.Management package, even though you removed the references to it. Before we were referencing version 7, now it's looking for 5. If 5 isn't in the feed, it makes sense that it wouldn't find it, but I'm not sure why it's even looking... There is still a `using System.Management` in there that I don't think should be, but I'm also having trouble building your branch locally right now. I'll keep poking around.

@krschau  - I was able to build locally without issue. Hmmm.  I added another push of some unused System.Management usings.

I DID find where the `System.Management 5.0.0` reference is coming from. It's `LibreHardwareMonitorLib` in `CoreWidgetProvider.csproj`

snickler commented:
Okay @EricJohnson327 @krschau, I also fixed the build with this. 

Had to do the following: 
- Add System.Management 7.0 NuGet package BACK in, because the package that has System.Management as a dependency is looking for an older version that is deprecated and seems to continuously fail.
- Updated the build workflow to use MSBuild for NuGet package restore. No more auth failures.

https://github.com/snickler/devhome/blob/66c0973b42576788d0190c1da6387e54ee1b4f38/.github/workflows/DevHome-CI.yml#L59-L62

![image](https://github.com/microsoft/devhome/assets/4016293/d880398f-a404-4e14-a3cc-d8135f3cac77)

krschau commented:
I don't think we actually need the LibreHardwareMonitorLib package? So probably that gets around System.Management 5.0.

I'd like @EricJohnson327 to look at the yaml changes.

snickler commented:
@krschau  - Yup, it builds without LibHardawreMonitor so I removed both of those packages. No more System.Management needed.

EricJohnson327 reviewed (approved):
I'm not an expert on the APIs or regkeys being used, so I'll leave that review to @krschau.  But the package dependency and yml changes look good to me.
